### PR TITLE
General Fixes for v1.3.rc

### PR DESCRIPTION
**Pull request by eksperimental**
- Remove double words (on on)
- Use capital letter for proper nouns such as: Elixir, Erlang, Mix, EEx, IEx, Dialyzer, Rebar, Make, Git, Markdown, Unix, Windows, etc.
- User proper casing for accronysm, such as: IO
- Remove use of em-dash(`—`) in favour of parenthesis
- Replace use of backticks from messages and comments, with double quotes
- Fix indentation in code examples
- Add missing arity in function mentions
- Add backticks where needed: typical uses case `true`, `false`, `nil`
- Remove space(s) between "fn" and "("
- Correct use of indefinite article a/an
- Convert hex to use uppercase after "x", ie: 0xffff to 0xFFFF, and \u{0fffe} to \u{0FFFE}
- Fix erlang.org to always be www.erlang.org
- Fix Mix error messages to start with capital letter,
- Fix all other raise message should startwith
- Fix module, function and task tasks to use tense proper tense.
- Fix wrongly closed backtick
- Always use https:// when available
- Use Markdown links instead of just displaying the plain URL.

UPDATE:
- Use proper case in Acronysms such as IPv6, WWW, HTTP, FTP, etc

### CODE REVIEW

**Comment by eksperimental** [comment]
there are changes in `bin/elixir.bat`, but that's because my editor converts end of lines to LF and strips trailing white spaces, 
Should we use CRLF for this .bat file? or what.

**Comment by whatyouhide** [comment]
I still have to review this thoroughly but I'm 👎 on one thing, which is double quotes instead of backticks in code comments. I agree that backticks are not ideal but I really don't see double quotes as an improvement. What if we just remove backticks? These comments are always seen in a monospaced font anyways, not by a Markdown processor. Wdyt?

Also, there are some changes in this PR which are objectively right (e.g., double words, capitalization) but others that are up for discussion. Have you thought about opening a PR for each of the fixes? It would be superquick to review and merge right away :)

**Comment by eksperimental** [comment]
Removing backticks sound good. I think we need to use them when it is ambiguous not having them.

Could you mention the ones that you think are up for discussion (or comment the code lines), i can revert until we reach an agreement.

**Comment by josevalim** [comment]
@eksperimental we really appreciate those improvements but, as previously asked, you need to break those PRs into smaller ones, because otherwise it becomes really hard to review, filter and give feedback.

File: lib/elixir/lib/kernel/utils.ex [hunk 1]
```diff
@@ -40,8 +40,7 @@ defmodule Kernel.Utils do
   end
 
   defp normalize_args(raw_args) do
-    :lists.foldr(fn
-      ({:\\, _, [arg, default_arg]}, [as_args | _] = as_args_list) ->
```
Comment (josevalim) [inline lib/elixir/lib/kernel/utils.ex]: Please revert as this function as multiple clauses.

File: lib/mix/lib/mix/tasks/xref.ex [hunk 2]
```diff
@@ -279,7 +279,7 @@ defmodule Mix.Tasks.Xref do
 
   defp raise_invalid_callee(callee) do
     message =
-      "xref --callers expects `Module`, `Module.function`, or `Module.function/arity`, got: " <>
+      "xref --callers expects \"Module\", \"Module.function\", or \"Module.function/arity\", got: " <>
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/xref.ex]: Great catch.

**Comment by josevalim** [comment]
About the backticks and double quotes in code comments, let's please keep the backticks. I believe we should write code comments as we write documentation, this also makes it easier if we eventually want to make some comment part of the documentation.

Thank you @eksperimental, I've fully reviewed it! 💯

**Comment by eksperimental** [comment]
I will break them into several PRs.

File: lib/mix/lib/mix/tasks/xref.ex (hunk 2 above)
Comment (eksperimental) [inline lib/mix/lib/mix/tasks/xref.ex]: do we need double quoted here? not having them is not confusing either

**Comment by eksperimental** [comment]
In the previous cleanup i did, we agreed on no backticks when inside comments.
https://github.com/elixir-lang/elixir/pull/3697#issuecomment-139200582

For me using backticks is more natural, but for some reason that was the choice.

**Comment by josevalim** [comment]
@eksperimental we have already reviewed this one, so no need to break it apart but please keep it in mind the next times. :) Also, regarding the backticks or double quotes, it seems I contradicted myself so I will go with whatever you and @whatyouhide decide.

File: lib/elixir/lib/macro.ex [hunk 3]
```diff
@@ -153,7 +153,7 @@ defmodule Macro do
     raise ArgumentError, bad_pipe(expr, call_args)
   end
 
-  # Without this, `Macro |> Env == Macro.Env`.
+  # Without this: "Macro |> Env == Macro.Env"
```
Comment (whatyouhide) [inline lib/elixir/lib/macro.ex]: Other than the quotes/backticks, please leave the trailing period in comments, as they're still sentences meant to be read by humans :)

**Comment by eksperimental** [comment]
I will be posting it in several PRs.
I think it's better like this.

**Comment by whatyouhide** [comment]
@eksperimental it's up to you of course 😃, but as @josevalim said we reviewed this (90% from me at least, but almost done 😛) so we could merge this one as well. Whatever you prefer! 💟

File: lib/elixir/lib/macro.ex (hunk 3 above)
Comment (eksperimental) [inline lib/elixir/lib/macro.ex]: Trailing period on single line comments? :-?

Comment (whatyouhide) [inline lib/elixir/lib/macro.ex]: Yep, the comment was meant to say that "Without this code, this happens". I would leave `,` and `.` there. (btw, I just remembered that I'm the one that wrote that comment! :)).

**Comment by eksperimental** [comment]
I think it is a really good point @josevalim , the ability to convert comments into a doc (from private function to public) without hassle.
So I think we should use backticks and not doublequotes inside comments

**Pull request closed without merging**